common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/extensions/types/PropertyExtension.java

```diff
@@ -0,0 +1,22 @@
+package io.vrap.rmf.codegen.common.generator.extensions.types;
+
+import io.vrap.rmf.codegen.common.processor.annotations.ExtensionMethod;
+import io.vrap.rmf.codegen.common.processor.annotations.ModelExtension;
+import io.vrap.rmf.raml.model.types.Property;
+import io.vrap.rmf.raml.model.types.StringInstance;
+import io.vrap.rmf.raml.model.types.StringType;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ModelExtension(extend = Property.class)
+public class PropertyExtension {
+
+
+    @ExtensionMethod
+    public Boolean  isPatternProperty(Property property){
```

Comment:
We once planned to add this directly to RMF, but I forgot why we haven't done it yet. I have created this issue for RMF https://github.com/vrapio/rest-modeling-framework/issues/25, but until its implemented we of course need this method :wink:

Reply:
actually thats the purpose is to make extending the existing model less painful 😄